nowriter/tmencoding/tm_encoder_facade_intr.go

```diff
@@ -0,0 +1,6 @@
+package tmencoding
+
+type TMEncoderFacadeIntr interface {
```

Comment:
it's the simplest, least sophisticated, most fool proof interface available in the system. it's marked 'Facade'. When this was written, the `Builder` version did not exist (and the Facade title may move over to it I think).  The main reason Facade is here now is because some people consider Builder chaining sophisticated and confusing. Whereas the style shown above is as simple as Builder and similar in statefulness but does not allow the chaining syntax so can be considered more pedestrian. It would be incomplete to not offer as an option during the larger refactoring and associated discussion. The strategy I am hoping we can use is have all reasonable options well named and separated and cleanly implemented here then we all can refactor to whichever preferred interface we like. Presumably we have one common one that most people use that is best. But in some odd cases we may want to use a variant and I don't want to hold up the larger refactoring by not having the correct tool available during the process.